### ISSUE TITLE
SU2 7.1.1 Develop Branch fails to build if using Pastix support 

### ISSUE THREAD

**Issue by Root-voidX** [su2code/SU2#1349]
**Describe the bug**

When compiling SU2 7.1.1 Blackbird from the Develop Branch with Pastix support, the build fails at Common/src/libSU2CommonAD.a.p/wall_model.cpp.o with a c++ error saying unrecognized command-line-option '-fexceptions;-pthread'.

Here is a snip from the error message: 
![image](https://user-images.githubusercontent.com/24353606/128623331-461f0ded-4b4d-4ece-83e8-b2eb40e9129d.png)

**To Reproduce**
Basically I download the develop branch from SU2 7.1.1,
extract it and configure using:
![image](https://user-images.githubusercontent.com/24353606/128623382-55f4bc2e-e971-4b1d-81bf-626fc9ae484d.png)

**Desktop (please complete the following information):**
 - OS: Fedora 34
 - C++ compiler and version: g++ (GCC) 11.2.1
 - MPI implementation and version: OpenMPI-4.1.1
 - SU2 Version: 7.1.1-Develop

**Comment by pcarruscag** [comment]
Mpi4 is not compatible with the version of pastix we support.
And you have to compile scotch and pastix according to the instructions in TestCases/pastix_support/ before compiling SU2

**Comment by Root-voidX** [comment]
> Mpi4 is not compatible with the version of pastix we support.
> And you have to compile scotch and pastix according to the instructions in TestCases/pastix_support/ before compiling SU2

Thanks for the clarification! I use Mpi4 for most of my programs so that's why I built it that way. I followed this guide for pastix: https://solverstack.gitlabpages.inria.fr/pastix/md_docs_doxygen_chapters_Pastix_Runtime.html

So I build pastix 6.X.X. I will check the test cases directory though for the instructions.  Also how does changing openmpi change c++ command line option ? 

Thanks 😊

**Comment by Root-voidX** [comment]
> Mpi4 is not compatible with the version of pastix we support.
> And you have to compile scotch and pastix according to the instructions in TestCases/pastix_support/ before compiling SU2

Just followed the pastix instructions from the testcase directory. compilation stops at 443/1058 saying error Cannot use PaStiX with forward mode AD.

Picture:
![image](https://user-images.githubusercontent.com/24353606/128684248-2aae9693-806b-421c-9c13-b2fedbcbe5f1.png)

So what am I missing? Also is PasTiX necessary for me to run/use all solvers needed for tutorials/test cases?

**Comment by pcarruscag** [comment]
It is useful in the circumstances listed here:
https://su2code.github.io/docs_v7/Build-SU2-Linux-MacOS/#linear-algebra-options
Not needed for tutorials.

**Comment by Root-voidX** [comment]
> It is useful in the circumstances listed here:
> https://su2code.github.io/docs_v7/Build-SU2-Linux-MacOS/#linear-algebra-options
> Not needed for tutorials.

Thanks for the clarification, so do you know why I got the error saying it cannot be used for forward mode AD? Is that a known limitation or am I missing something ?

**Comment by pcarruscag** [comment]
Because it can't be used, we don't support it.

**Comment by Root-voidX** [comment]
@pcarruscag 
[config-and-makefile-for-scotch-and-pastix.zip](https://github.com/su2code/SU2/files/6959270/config-and-makefile-for-scotch-and-pastix.zip)

So very good news, I compiled successfully on Fedora 34 using GCC-11.2 and OpenMPI-4.1.1 using Scotch-6.0.6 and Pastix-5.2.3. I have attached the relevant config and make-file with my relevant edits for reference. Newer openmpi works if you are willing to edit the CCFLAGS pointing to their bin, lib and include file. Currently had no issues running test cases such as fea and aeroelastic. However I don't know which one specifically uses pastix so i can test that but if you do know, let me know. I disabled forward mode ad during meson config so I didn't run into the error. I also enabled shared libraries for both scotch and pastix. 

Looking forward  to your response =)
